File: ad_library/src/main/java/com/adgrowth/internal/exceptions/APIIOException.java
package com.adgrowth.internal.exceptions;


import com.adgrowth.internal.http.APIClient;

import org.json.JSONException;
import org.json.JSONObject;


public class APIIOException extends Exception {
    private String mMessage;

    private JSONObject body;
    private int statusCode;

    public APIIOException(APIClient.Response response) {
        this.statusCode = response.getStatusCode();
        this.mMessage = response.getMessage();

        try {
            body = response.json();

            if (body != null && body.has("message"))
                mMessage = (String) body.get("message");

        } catch (JSONException ignored) {
        }
    }

    public APIIOException(int statusCode, String message) {
        this.statusCode = statusCode;
        this.mMessage = message;
        body = new JSONObject();

    }

    @Override
    public String getMessage() {
        return mMessage;
    }

    public JSONObject getBody() {
        return body;
    }

    public int getStatusCode() {
        return statusCode;
    }
}

File: ad_library/src/main/kotlin/com/adgrowth/internal/exceptions/APIIOException.kt
package com.adgrowth.internal.exceptions

import com.adgrowth.internal.http.HTTPStatusCode
import com.adgrowth.internal.http.HttpClient
import org.json.JSONException
import org.json.JSONObject

class APIIOException : Exception {
    override var message: String?
        private set
    var body: JSONObject = JSONObject()
        private set
    var statusCode: Int
        private set

    constructor(response: HttpClient.Response) {
        statusCode = response.statusCode
        message = response.message

        try {
            body = response.json()
            if (body.has("message")) message = body["message"] as String
        } catch (ignored: JSONException) {
        }
    }

    constructor(statusCode: Int, message: String) {
        this.statusCode = statusCode
        this.message = message
        body = JSONObject()
    }
}
